### PR TITLE
Improving the toggle logic for disabling controllers from the kafka control plane 

### PR DESCRIPTION
**Pull request by matzew**

### CODE REVIEW

File: knative-operator/pkg/common/util.go [hunk 1]
```diff
@@ -17,6 +17,26 @@ import (
...
+
+// Gets StringMap values as comma separated string
+func (m StringMap) StringValues() string {
+	values := make([]string, 0, len(m))
+
+	for _, v := range m {
+		values = append(values, v)
+	}
+
+	return strings.Join(values[:], ",")
```
Comment (pierDipi) [inline knative-operator/pkg/common/util.go:37]: I think there is no need for this slicing, right?
```suggestion
	return strings.Join(values, ",")
```

**Comment by matzew** [comment]
@pierDipi updated and added a bit of tests

File: knative-operator/pkg/controller/knativekafka/knativekafka_controller.go [hunk 2]
```diff
@@ -501,22 +501,29 @@ func configureEventingKafka(spec serverlessoperatorv1alpha1.KnativeKafkaSpec) mf
...
 
-			if spec.Broker.Enabled && spec.Sink.Enabled {
-				// all: nothing to disable
-				deployment.Spec.Template.Spec.Containers[0].Args = nil
-			} else if spec.Broker.Enabled {
-				// only Broker: we disable the sink controllers
-				deployment.Spec.Template.Spec.Containers[0].Args = []string{"--disable-controllers=sink-controller"}
-			} else if spec.Sink.Enabled {
+			if spec.Broker.Enabled {
+				// broker is enabled, so we remove alits controllers:
```
Comment (pierDipi) [inline knative-operator/pkg/controller/knativekafka/knativekafka_controller.go:515]: alits?

Comment (matzew) [reply knative-operator/pkg/controller/knativekafka/knativekafka_controller.go]: :smile:  all of its

**Comment by mgencur** [comment]
It's a coincidence that I have just filed this one: https://issues.redhat.com/browse/SRVCOM-1630 . But it's a bit related.

**Review by pierDipi — APPROVED** [review]
/lgtm

**Pull request merged**